models/issue_tracked_time.go

```diff
@@ -24,20 +24,49 @@ type TrackedTime struct {
 	Time        int64     `json:"time"`
 }
 
+// TrackedTimeList is a List ful of TrackedTime's
+type TrackedTimeList []*TrackedTime
+
 // AfterLoad is invoked from XORM after setting the values of all fields of this object.
 func (t *TrackedTime) AfterLoad() {
 	t.Created = time.Unix(t.CreatedUnix, 0).In(setting.DefaultUILocation)
 }
 
 // APIFormat converts TrackedTime to API format
 func (t *TrackedTime) APIFormat() *api.TrackedTime {
+	user, err := GetUserByID(t.UserID)
+	if err != nil {
+		return nil
+	}
+	issue, err := GetIssueByID(t.IssueID)
+	if err != nil {
+		return nil
+	}
+	err = issue.LoadRepo()
+	if err != nil {
+		return nil
+	}
 	return &api.TrackedTime{
-		ID:      t.ID,
-		IssueID: t.IssueID,
-		UserID:  t.UserID,
-		Time:    t.Time,
-		Created: t.Created,
+		ID:       t.ID,
+		IssueID:  t.IssueID,
+		Issue:    issue.APIFormat(),
+		UserID:   t.UserID,
+		UserName: user.Name,
+		Time:     t.Time,
+		Created:  t.Created,
+	}
+}
+
+// APIFormat converts TrackedTimeList to API format
+func (tl TrackedTimeList) APIFormat() api.TrackedTimeList {
+	var result api.TrackedTimeList
```

Comment:
You know how long tl is so you know how long your result is. Preallocate.

Reply:
done